components/ui/CopyButton.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import { useState } from "react";
+import { BoxButton } from "./BoxButton";
+import CopyIcon from "public/assets/icons/copy.svg";
+import CopyActiveIcon from "public/assets/icons/copy_active.svg";
+
+export const CopyButton = ({ content, children, compact, classes }: {content: string, children?: any, compact?: boolean, classes?: string }) => {
+    const [copied, setCopied] = useState(false);
+
+    function copy() {
+        const el = document.createElement('textarea');
+        el.value = content;
+        document.body.appendChild(el);
+        el.select();
+        document.execCommand('copy');
+        document.body.removeChild(el);
+
+        setCopied(true);
+        setTimeout(() => setCopied(false), compact ? 800 : 2000);
+    }
+
+    if (compact) {
+        return (
+            <button className={`${classes} cursor-pointer`} onClick={copy}>
```

Comment:
The button needs a title, so screen readers know what it does. Could be `Copy` by default, and change to `Copied` when clicked.

When clicked, it should also have `aria-pressed="true"`. An attribute we should also add to the `BoxButton` at some point.

Reply:
Done